resources/setup_emulator.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -e
+
+kill_all_emus() {
+    for emu_device in $(adb devices -l |grep 'device product:' |cut -d' ' -f1); do
+        echo "Killing emulator: $emu_device"
+        adb -s $emu_device emu kill
+    done
+}
+
+echo 'Creating emulator'
+echo 'n' | $ANDROID_HOME/tools/bin/avdmanager create avd -n test-android -f -k "system-images;android-27;google_apis;x86" 
+
+echo 'loading emulator'
+$ANDROID_HOME/emulator/emulator -avd test-android -no-boot-anim -no-window -noaudio & 
+sleep 2m
+
+EXPECTED_DEVICE_NAME='emulator-5554'
+$ANDROID_HOME/platform-tools/adb devices |grep $EXPECTED_DEVICE_NAME
+if [ $? == 0 ]; then
+   echo "Sending Key event, to press HOME button"
+   $ANDROID_HOME/platform-tools/adb shell input keyevent 3 &
```

Comment:
^ What is the purpose of this step?

Reply:
It is pressing HOME button of device, useful to make sure if device is responsive

Reply:
Since it can check if the device is responsive, does it throw an error or alert us if it is not responsive? Should we account for that?

Reply:
It will show error if device is not responsive, added a check on it as well.